x-pack/plugins/observability_ai_assistant/public/hooks/use_observability_ai_assistant_router.ts

```diff
@@ -43,6 +50,9 @@ export function useObservabilityAIAssistantRouter(): StatefulObservabilityAIAssi
 
         history.push(next);
       },
+      navigateToConversationsApp: (path, ...args) => {
+        navigateToApp('observabilityAIAssistant', { path, ...args });
+      },
```

Comment:
I suggest keeping the `router` api simple and using `push`/`replace` directly. 

```suggestion
```

Reply:
See earlier comment

Reply:
In order to do so we have to change `push` and `replace` to use `navigateToApp`:

```ts
      push: (...args) => {
        const next = link(...args);
        navigateToApp('observabilityAIAssistant', { path: next, replace: false });
      },
      replace: (path, ...args) => {
        const next = link(path, ...args);
        navigateToApp('observabilityAIAssistant', { path: next, replace: true });
      },
```

Reply:
But then we would also use the navigateToApp method when it is not needed, like when we are in the conversations app.

Reply:
> But then we would also use the navigateToApp method when it is not needed, like when we are in the conversations app.

Is that an actual problem?

Reply:
We could also have two pairs of `push`/`replace` methods. One pair for navigating within the app (current ones) and the other pair for navigating from another app into the AI Assistant app. But I am not sure if it's actually necessary.

Reply:
Not sure if it's an actual problem. FWIW, I always found it slightly odd that Kibana has a separate `navigate` and `navigateToApp` method. Unsure as to why that is, do you know?

Reply:
No, I don't either

Reply:
I've tried using `replace` from the existing router API:

```
      replace: (path, ...args) => {
        const next = link(path, ...args);
        history.replace(next);
      },
      link: (path, ...args) => {
        return http.basePath.prepend('/app/observabilityAIAssistant' + link(path, ...args));
      },
```

it really does not navigate the user to a different app, even though it uses `link` with the `http.basePath.prepend` etc.

It seems we need a dedicated method when navigating from another app into the conversations app.

Reply:
Implemented b03cd4ddedac10b3e347abfcfbf9fc38cab8ba29